src/bin/languageserver/mod.rs

```diff
@@ -1179,6 +1179,8 @@ impl LanguageServer for SolangServer {
         if let Ok(path) = uri.to_file_path() {
             self.files.lock().await.remove(&path);
         }
+        
+        self.client.publish_diagnostics(uri, vec![], None).await;
```

Comment:
Wonderful!

Again, this could do with a test

Reply:
I tried writing a test using the `vscode` typescript API to open and close files. 
I am using `vscode.commands.executeCommand('workbench.action.closeActiveEditor')` to close the open tab. But this doesn't seem to emit the `didClose` event which is what the language server listens for. 
I tried registering a listener for this event in javascript as well. But, this listener did not run either. 
But manually closing the files works every time and the old diagnostics disappear when this happens.